app/utils.py

```diff
@@ -0,0 +1,193 @@
+    params = {'city_id': loc_id}
+
+    response = requests.get(url, headers=headers, params=params)
+
+    cuisines = []
+
+    if response:
+        body = response.json()
+        for cuisine in body['cuisines']:
+            cuisines.append(cuisine['cuisine']['cuisine_name'])
+
+    return cuisines
+
+def get_cuisine_id(loc_id, cuisine_names):
+    """
+    Queries Zomato API for cuisine id based off the passed in cuisine
+    """
+
+    url = 'https://developers.zomato.com/api/v2.1/cuisines'
+    headers = {'user_key': 'e74153df38632880474d4788e0936560'}
+    params = {'city_id': loc_id}
+
+    response = requests.get(url, headers=headers, params=params)
+
+    ids = []
+
+    if response:
+        body = response.json()
+        for cuisine in body['cuisines']:
+            if cuisine['cuisine']['cuisine_name'] in cuisine_names:
+                ids.append(cuisine['cuisine']['cuisine_id'])
+
+    return ids
+
+def find_establishments(loc_id):
+    """
+    Queries Zomato API for estalishment types at location id
+    """
+
+    url = 'https://developers.zomato.com/api/v2.1/establishments'
+    headers = {'user_key': 'e74153df38632880474d4788e0936560'}
+    params = {'city_id': loc_id}
+
+    response = requests.get(url, headers=headers, params=params)
+
+    establishments = []
+
+    if response:
+        body = response.json()
+        for establishment in body['establishments']:
+            establishments.append(establishment['establishment']['name'])
+
+    return establishments
+
+def get_establishment_id(loc_id, establishment_names):
+    """
+    Queries Zomato API for establishment types at location id
+    """
+
+    url = 'https://developers.zomato.com/api/v2.1/establishments'
+    headers = {'user_key': 'e74153df38632880474d4788e0936560'}
+    params = {'city_id': loc_id}
+
+    response = requests.get(url, headers=headers, params=params)
+
+    ids = []
+
+    if response:
+        body = response.json()
+        for establishment in body['establishments']:
+            if establishment['establishment']['name'] in establishment_names:
+                ids.append(establishment['establishment']['id'])
+
+    return ids
+
+def search_restaurants(loc_id, cat_ids, cu_ids, establ_ids):
+    url = 'https://developers.zomato.com/api/v2.1/search'
+    headers = {'user_key': 'e74153df38632880474d4788e0936560'}
+    params = {'entity_id': loc_id, 'cuisine': list_to_string(cu_ids), 'establishment_type': list_to_string(establ_ids), 'category': list_to_string(cat_ids), 'entity_type': 'city'}
+
+    response = requests.get(url, headers=headers, params=params)
+    
+    ids = []
+
+    if response:
+        body = response.json()
+        for restaurant in body['restaurants']:
+            ids.append(restaurant['restaurant']['R']['res_id'])
+    
+    return ids
+
+def get_restaurant_details(res_id):
+    url = 'https://developers.zomato.com/api/v2.1/restaurant'
+    headers = {'user_key': 'e74153df38632880474d4788e0936560'}
```

Comment:
I'm not sure what this headers part is doing, a comment or something would be nice.

Reply:
that's from zomato, pretty typical for accessing public apis